tools/src/main/scala/com/nvidia/spark/rapids/tool/planparser/SQLPlanParser.scala

```diff
@@ -285,7 +285,62 @@ object SQLPlanParser extends Logging {
         case _ => // NO OP
       }
     }
-    parsedExpressions.toArray
+    parsedExpressions.distinct.toArray
+  }
+
+  // This parser is used for SortAggregateExec, HashAggregateExec and ObjectHashAggregateExec
+  def parseAggregateExpressions(exprStr: String): Array[String] = {
+    val parsedExpressions = ArrayBuffer[String]()
+    // (key=[num#83], functions=[partial_collect_list(letter#84, 0, 0), partial_count(letter#84)])
+    val pattern = """functions=\[([\w#, +*\\\-\.<>=\`\(\)]+\])""".r
+    val aggregatesString = pattern.findFirstMatchIn(exprStr)
+    // This is to split multiple column names in AggregateExec. Each column will be aggregating
+    // based on the aggregate function. Here "partial_" is removed and only function name is
+    // preserved. Below regex will first remove the "functions=" from the string followed by
+    // removing "partial_". That string is split which produces an array containing
+    // column names. Finally we remove the parentheses from the beginning and end to get only
+    // the expressions. Result will be as below.
+    // paranRemoved = Array(collect_list(letter#84, 0, 0),, count(letter#84))
```

Comment:
Since I am doing a regex lookbehind on the string `),` the resultant array will be
[partial_collect_list(letter#84, 0, 0),
count(letter#84)]

First `,` is from the first array element followed by separator(`,`). Will update this if necessary based on your suggestion in below comment.